lib/ecto/repo/model.ex

```diff
@@ -181,7 +201,8 @@ defmodule Ecto.Repo.Model do
   end
 
   defp with_transactions_if_callbacks(repo, adapter, model, opts, callbacks, fun) do
-    if Enum.any?(callbacks, &function_exported?(model, &1, 1)) do
+    if Enum.any?(callbacks, &function_exported?(model, &1, 1)) and
```

Comment:
Sorry, you need to remove the function definition. Otherwise this code will be triggered anyway.